### PR TITLE
CIRCUITPY_DEFAULT_WIFI_TX_POWER

### LINKED ISSUE #9235: Wemos C3 Mini should pre-set tx_power to 8.5
### CircuitPython version

```python
main
```

### Code/REPL

```python
n/a
```

### Behavior

As reported by multiple users, with the power-on default tx_power, the board does not successfully connect to wifi (https://github.com/adafruit/circuitpython/issues/6540 https://github.com/adafruit/circuitpython/issues/9058#issuecomment-2104612719 and probably others) but does work if boot.py executes `wifi.radio.tx_power=8.5`.

### Description

_No response_

### Additional information

The board init routine should set tx_power=8.5 by default, without needing lines in boot.py.

### PR DESCRIPTION
**Pull request by bill88t**
Alternative to #9357.
Fixes #9235.

### CODE REVIEW

File: ports/espressif/common-hal/wifi/Radio.c [hunk 1]
```diff
@@ -86,6 +86,9 @@ void common_hal_wifi_radio_set_enabled(wifi_radio_obj_t *self, bool enabled) {
     if (!self->started && enabled) {
         ESP_ERROR_CHECK(esp_wifi_start());
         self->started = true;
+        #if CIRCUITPY_REDUCE_TX_POWER
+        common_hal_wifi_radio_set_tx_power(self, 11.0);
+        #endif
```
Comment (dhalbert) [inline ports/espressif/common-hal/wifi/Radio.c:91]: This is tricky. The C preprocessor can't dp `#if` with floating constants. So in order to conditionalize this so the `common_hal_wifi_radio_set_tx_power()` is not compiled if the power is not specified, we use a constant expression which will be optimized.
```suggestion
        if (CIRCUITPY_WIFI_DEFAULT_TX_POWER != 20) {
            common_hal_wifi_radio_set_tx_power(self, CIRCUITPY_WIFI_DEFAULT_TX_POWER);
        }
```

File: ports/espressif/mpconfigport.h [hunk 2]
```diff
@@ -59,3 +59,9 @@
 extern portMUX_TYPE background_task_mutex;
 #define CALLBACK_CRITICAL_BEGIN (taskENTER_CRITICAL(&background_task_mutex))
 #define CALLBACK_CRITICAL_END (taskEXIT_CRITICAL(&background_task_mutex))
+
+// Reduce wifi radio power
+// Only for boards with imperfect radio designs
+#ifndef CIRCUITPY_REDUCE_TX_POWER
+#define CIRCUITPY_REDUCE_TX_POWER (0)
+#endif
```
Comment (dhalbert) [inline ports/espressif/mpconfigport.h:67]:
```suggestion
// 20 dBm is the default and the highest max tx power.
// Allow a different value to be specified for boards that have trouble with using the maximum power.
#ifndef CIRCUITPY_WIFI_DEFAULT_TX_POWER
#define CIRCUITPY_WIFI_DEFAULT_TX_POWER (20)
#endif
```

**Review by dhalbert — CHANGES_REQUESTED** [review]
How about a `CIRCUITPY_DEFAULT_WIFI_TX_POWER`, so that different values can be set for different boards? I've included change suggestions.

**Comment by bill88t** [comment]
I'll test tomorrow, thanks!

**Comment by bill88t** [comment]
If we wanted to keep the check we should do a `fabs(20 - default) > 0.1`, but the check is not really needed.

Tested on cardputer and supertiny.

File: ports/espressif/common-hal/wifi/Radio.c [hunk 3]
```diff
@@ -86,9 +86,7 @@ void common_hal_wifi_radio_set_enabled(wifi_radio_obj_t *self, bool enabled) {
     if (!self->started && enabled) {
         ESP_ERROR_CHECK(esp_wifi_start());
         self->started = true;
-        if (CIRCUITPY_WIFI_DEFAULT_TX_POWER != 20) {
-            common_hal_wifi_radio_set_tx_power(self, CIRCUITPY_WIFI_DEFAULT_TX_POWER);
-        }
+        common_hal_wifi_radio_set_tx_power(self, CIRCUITPY_WIFI_DEFAULT_TX_POWER);
```
Comment (dhalbert) [inline ports/espressif/common-hal/wifi/Radio.c:89]: The check removes an unnecessary call at compile time (assuming optimizer is working). It was previously done with #if in your original PR.

**Review by dhalbert — CHANGES_REQUESTED** [review]

File: ports/espressif/common-hal/wifi/Radio.c (hunk 3 above)
Comment (dhalbert) [reply ports/espressif/common-hal/wifi/Radio.c:89]: Well, maybe never mind, since it could be that in the future some other default was chosen.

**Review by dhalbert — APPROVED** [review]

File: ports/espressif/common-hal/wifi/Radio.c (hunk 3 above)
Comment (bill88t) [reply ports/espressif/common-hal/wifi/Radio.c:89]: Oh I didn't know the optimizer caught these things. I'll keep it in mind next time.

**Pull request merged**